src/chainlit/server.py

```diff
@@ -160,17 +161,19 @@ def get_html_template():
     <meta property="og:image" content="https://chainlit-cloud.s3.eu-west-3.amazonaws.com/logo/chainlit_banner.png">
     <meta property="og:url" content="{url}">"""
 
+    js = f"""<script>
+    window._headers = {headers};
+    </script>"""
```

Comment:
looks really insecure to me, that means that any script we load can access the initial headers such as Authorization

Reply:
why not put the headers in the user session on the back end, instead?

Reply:
I agree this was the first idea I had. Maybe we should put it in a cookie, way safer. The use case is I have a chainlit app running in a company secured network. I have a proxy in front of the app using my company auth and I then redirect to the chainlit app with a custom auth header. I then want to be able to access those headers as a developer.

Reply:
but why do we need to do anything on the front end? if I have a proxy in front of my app, all the requests I send to the app are going to have my custom headers

Reply:
because we need to access them in the websocket connection also. I just pushed a change to remove the headers and  just store them in a cookie. Socket io also sends the cookies.

Reply:
AFAIK there's no reason that the websocket HTTP handshake request wouldn't have the proxy headers as well. Am I missing something?